flags/src/test/java/org/cloudname/flags/FlagsTest.java

```diff
@@ -372,4 +378,102 @@ public void testLoadingFromMultiplePropertyFiles2() throws Exception {
         Assert.assertNull(FlagsPropertiesFile.comments);
     }
 
+    static class BaseFlaggedTestClass {
+        static int baseCallCounter;
+
+        @PostConstruct
+        protected void b0() {baseCallCounter++;}
+
+        @PostConstruct
+        private void b1() {baseCallCounter++;}
+
+    }
+    static class FlaggedTestClass extends BaseFlaggedTestClass {
+        @Flag (name="flag-one")
+        private String flagOne;
+
+        static int callCounter;
+
+        @PostConstruct
+        public void i1() { callCounter++; }
+
+        @PostConstruct
+        private void i2() {  callCounter++; }
+
+        @PostConstruct
+        protected void i3() { callCounter++;}
+
+        @PostConstruct
+        static void i4() {
+            callCounter++;
+        }
+    }
+
+    static class FlaggedTestStaticClass {
+        @Flag (name="flag-two")
+        private static String flagTwo;
+        public static int callCounter;
+
+        @PostConstruct
+        public static void c1() { callCounter++; }
+
+        // check that instance methods are ignored
+        @PostConstruct
+        public void i2() { callCounter++; }
+
+    }
+
+    @Test
+    public void testPostConstructForInstanceConfiguration() {
+        final String[] args = new String[] { "--flag-one", "xyz"};
+        final FlaggedTestClass instance = new FlaggedTestClass();
+        new Flags().loadOpts(instance).parse(args);
+
+        assertThat(instance.flagOne, is("xyz"));
+        assertThat(FlaggedTestClass.callCounter, is(3));
+
+        // check that we don't support inherited methods for now
+        assertThat(BaseFlaggedTestClass.baseCallCounter, is(0));
+    }
+
+    @Test
+    public void testPostConstructForСlassConfiguration() {
+        final String[] args = new String[] { "--flag-two", "abc"};
+        new Flags().loadOpts(FlaggedTestStaticClass.class).parse(args);
+
+        assertThat(FlaggedTestStaticClass.flagTwo, is("abc"));
+        assertThat(FlaggedTestStaticClass.callCounter, is(1));
+        // don't support i methods.
```

Comment:
I really didn't understand "i methods". Since you have methods called "i1", "i2", etc, I naturally assumed you were referring to those as "i methods". That's why I assumed you actually meant "b methods", since the methods starting with "b" are the inherited methods. Please change "i" to "inherited" here to avoid the confusion.